Sources/BarcodeHeroUI/Views/BHFocusAreaView.swift

```diff
@@ -0,0 +1,113 @@
+        private static let labelHeight: CGFloat = 40
+        private static let width: CGFloat = 255
+
+        // MARK: Properties - Views
+
+        private lazy var barcodeDataLabel: UILabel = {
+            let dataLabel = UILabel()
+            dataLabel.textAlignment = .center
+            dataLabel.translatesAutoresizingMaskIntoConstraints = false
+            dataLabel.heightAnchor.constraint(equalToConstant: Self.labelHeight).isActive = true
+
+            return dataLabel
+        }()
+
+        private lazy var barcodeTypeLabel: UILabel = {
+            let typeLabel = UILabel()
+            typeLabel.textAlignment = .center
+            typeLabel.translatesAutoresizingMaskIntoConstraints = false
+            typeLabel.heightAnchor.constraint(equalToConstant: Self.labelHeight).isActive = true
+
+            return typeLabel
+        }()
+
+        private(set) lazy var cutoutView: UIView = {
+            let cutoutView = UIView()
+            cutoutView.translatesAutoresizingMaskIntoConstraints = false
+            cutoutView.heightAnchor.constraint(equalToConstant: Self.cutoutHeight).isActive = true
+            cutoutView.widthAnchor.constraint(equalToConstant: Self.width).isActive = true
+
+            return cutoutView
+        }()
+
+        // MARK: Properties - Convenience
+
+        var barcodeData: String? {
+            get {
+                return self.barcodeDataLabel.text
+            }
+            set {
+                self.barcodeDataLabel.text = newValue
+            }
+        }
+
+        var barcodeType: String? {
+            get {
+                return self.barcodeTypeLabel.text
+            }
+            set {
+                self.barcodeTypeLabel.text = newValue
+            }
+        }
+
+        // MARK: Methods - Lifecycle
+        
+        init() {
+            super.init(frame: .zero)
+            
+            self.translatesAutoresizingMaskIntoConstraints = false
+
+            let stackView = UIStackView()
+            stackView.axis = .vertical
+            stackView.translatesAutoresizingMaskIntoConstraints = false
+            stackView.spacing = 8
+
+            stackView.addArrangedSubview(self.barcodeDataLabel)
+            stackView.addArrangedSubview(self.cutoutView)
+            stackView.addArrangedSubview(self.barcodeTypeLabel)
+
+            self.addSubview(stackView)
+
+            let totalSpacing = stackView.spacing * CGFloat(stackView.arrangedSubviews.count - 1)
+            let height = (Self.labelHeight * 2) + Self.cutoutHeight + totalSpacing
+
+            // Activate height and width constraints for this view
+            NSLayoutConstraint.activate([
+                self.heightAnchor.constraint(equalToConstant: height),
+                self.widthAnchor.constraint(equalToConstant: Self.width),
+            ])
+
+            // Activate pinning constraints for the embedded stack view
+            NSLayoutConstraint.activate([
```

Comment:
any reason these are in two separate activate calls?

Reply:
Just wanted to group them more appropriately since the top group is for `self` constraints and this group is for `stackView` constraints against this view.

I don't believe there's any optimization in grouping them, but I can group them if you think it's better and just put comments inline with the array?

Reply:
`Typically, using this method is more efficient than activating each constraint individually.` That makes me thinnnkk that 2 activate calls is less efficient than one bigger one.  I guess I would group them all in one and arrange them accordingly, or add spaces or comments inside the array to make grouping more clear.